File: src/main/scala/com/backfield/imagegenerator/scala/sep2014/Color.scala
package com.backfield.imagegenerator.scala.sep2014

object Color {

  val baseColor = 0x313A75

  val baseColorPalette = List(0x182157, 0x080F3A, 0x525B92, 0x7C93AF)

  val compliment = 0xAA8A39

  val complimentPalette = List(0xFFE7AA, 0xD4B66A, 0x806115, 0x553D00)

  def nextColor : Int = {
    val rand = Randomness().nextFloat()
    val (base, palette) = if(rand < 0.1) {
      (compliment, complimentPalette)
    } else {
      (baseColor, baseColorPalette)
    }
    if(rand < 0.6) {
      base
    } else {
      palette((rand * 4).toInt)
    }
  }

}

File: src/main/java/com/backfield/imagegenerator/java/sep2014/Color.java
package com.backfield.imagegenerator.java.sep2014;

import java.security.SecureRandom;
import java.util.Random;

public class Color {

    int baseColor = 0x313A75;

    int[] baseColorPalette = {0x182157, 0x080F3A, 0x525B92, 0x7C93AF};

    int compliment = 0xAA8A39;

    int[] complimentPalette = {0xFFE7AA, 0xD4B66A, 0x806115, 0x553D00};

    public int nextColor(Random random) {
        float rand = random.nextFloat();
        int base;
        int[] palette;
        if(rand < 0.1) {
            base = compliment;
            palette = complimentPalette;
        } else {
            base = baseColor;
            palette = baseColorPalette;
        }
        if(rand < 0.6) {
            return base;
        } else {
            return palette[(int)(rand * 4)];
        }
    }
}
